### PR TITLE
feat: Add gcs filesystem

### PR DESCRIPTION
**Pull request by biellls**
Added GCS backend to the filesystem interface in Airflow Tools.

### CODE REVIEW

File: src/airflow_tools/filesystems/filesystem_factory.py [hunk 1]
```diff
@@ -25,6 +34,9 @@ def get_data_lake_filesystem(connection: Connection) -> FilesystemProtocol:
         elif connection.conn_type == "aws":
             hook = S3Hook(aws_conn_id=connection.conn_id)
             return S3Filesystem(hook)
+        elif connection.conn_type == 'google_cloud_platform':
```
Comment (victor-centelles-dkl) [inline src/airflow_tools/filesystems/filesystem_factory.py:37]: Just curious, Why do we use `aws`  for `Amazon Web Services` but for `Google Cloud Platform` we don't use `gcp`?

File: tests/integration/test_gcs_filesystem.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+from airflow.hooks.base import BaseHook
+
+from airflow_tools.filesystems.filesystem_factory import FilesystemFactory
+from airflow_tools.filesystems.impl.google_cloud_storage_filesystem import GCSFilesystem
+
+
+def test_gcs_filesystem():
+    """Expects an environment variable set to a GCS test bucket.
+    AIRFLOW_CONN_GCP_DATA_LAKE_TEST='{"conn_type": "google_cloud_platform", "extra": {"key_path": "/.../keyfile.json"}}'"""
```
Comment (victor-centelles-dkl) [inline tests/integration/test_gcs_filesystem.py:11]: I assume we have an static bucket for this test in GCS, right?

**Review by victor-centelles-dkl — APPROVED** [review]
LGTM

File: src/airflow_tools/filesystems/filesystem_factory.py (hunk 1 above)
Comment (biellls) [reply src/airflow_tools/filesystems/filesystem_factory.py:37]: We have to use the `conn_type` of the hook we are using:
[See AWS](https://github.com/apache/airflow/blob/main/providers/src/airflow/providers/amazon/aws/hooks/base_aws.py#L476)
[See GCS](https://github.com/apache/airflow/blob/main/providers/src/airflow/providers/google/common/hooks/base_google.py#L218)

File: tests/integration/test_gcs_filesystem.py (hunk 2 above)
Comment (biellls) [reply tests/integration/test_gcs_filesystem.py:11]: I have used a static bucket that we have but I think it's not in Pulumi. We should probably create an official one.

**Pull request merged**